## 4. Anthocyanin production and phenylpropanoid pathway in lingonberry
As the last part of my thesis, it would be nice to highlight the presence of important enzymatic pathways in anthocyanin biosynthesis (phenylpropanoid pathway, flavonoid pathway) in berries. \
My goal is then to show the presence of those genes in my assembly, hopefully quantify them in different tissue types (various berry and flower development stages).

How do you do it? 
I have: 
- Annotation file in .gff3 
- Annotated genes in .fasta, .faa where the feature name corresponds to each gene
- EggNog mapper results for orthologous genes in model organisms, labelled with KEGG, GO terms, etc. 
- RNAseq data from different tissue types


Example papers: 
- Blueberry paper (Colle, et al. 2019) is a good example showing the panel of flavonoid biosynthetic pathways in berries at different developmental stages (Figure 3). 
- Subtropical blueberry paper (Cui et al. 2022) performed GO enrichment analysis and tissue-specific expression profiles (Figure 6). Cuticle formation pathway is highlighted for similar type of analysis (Figure 7). 
- Another blueberry paper (Yu et al. 2019) is a good example showing flavonoid biosynthesis pathways in berries, directly highlighting the differentially expressed genes (DEGs) identified by different berry developmental stages on KEGG map (Figure 5).



